### PR TITLE
custom search for strings in tests

### PR DESCRIPTION
**Pull request by algojack**
## Summary
Brice's ticket so maybe he can expand on this a bit:
We need to find where we are missing 'testpartitioning.PartitionTest(t)' in our tests (files ending in "_test.go" in functions starting with "Test"), so this is a custom linter designed to do just that and can be expanded to do more.

## Test Plan
Tested on local (mac arm64) manually

### CODE REVIEW

File: scripts/custom_search.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+            # If previous line had search_string_one
+            if found_first_line:
+                if search_string_two not in line:
+                    print("WARNING: Line {} in file {} is missing '{}' in '{}'".format(index+1, filepath, search_string_two, found_first_line.strip()))
+                    results_found += 1
+                found_first_line = False
+
+            # If current line has search_string_one
+            elif search_string_one in line:
```
Comment (bricerisingalgorand) [inline scripts/custom_search.py:44]: I think this should be a regex expression right? For the linting we're trying to do I think we'll need to use something like `func\ ([A-Z])\w+\(t\ \*testing.T\)\ {`

Comment (algojack) [reply scripts/custom_search.py]: updated

**Comment by algojack** [comment]
Please DO NOT MERGE, we are exploring other options

**Comment by algojack** [comment]
sample output:
```
...
/Users/barbara/Documents/Projects/circle-poc/jack/go-algorand/test/netperf-go/puppeteer/puppeteer_test.go:23:1: TestMetricsPrintout function is missing testpartitioning.PartitionTest(<T type parameter>)
/Users/barbara/Documents/Projects/circle-poc/jack/go-algorand/util/condvar/timedwait_test.go:29:1: TestTimedWaitSignal function is missing testpartitioning.PartitionTest(<T type parameter>)
/Users/barbara/Documents/Projects/circle-poc/jack/go-algorand/util/condvar/timedwait_test.go:53:1: TestTimedWaitBroadcast function is missing testpartitioning.PartitionTest(<T type parameter>)
/Users/barbara/Documents/Projects/circle-poc/jack/go-algorand/util/condvar/timedwait_test.go:77:1: TestTimedWaitTimeout function is missing testpartitioning.PartitionTest(<T type parameter>)
exit status 3
```

**Comment by algojack** [comment]
closing this one as it will go straight to upstream master

**Pull request closed without merging**